course7/homeworks/anton/ls/main.cc

```diff
@@ -0,0 +1,135 @@
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+struct configuration
+{
+    std::string file_path = ".";
+    bool show_help = false;
+    bool show_tree = false;
+    bool show_long = false;
+};
+
+configuration parse_args(int argc, char *argv[]);
+void print_file_tree(const fs::path& dir, const configuration& config, const std::string& filler = "");
+size_t get_file_size(const fs::path&);
+
+int main(int argc, char *argv[])
+{
+    try
+    {
+        const configuration& config = parse_args(argc, argv);
+
+        if (!config.show_help)
+        {
+            fs::path path(config.file_path);
+            print_file_tree(path, config);
+        }
+
+        return 0;
+    }
+    catch(std::exception& ex)
+    {
+        std::cout << "\033[1;31m" << ex.what() << "\033[0m" << std::endl;
+        return 1;
+    }
+}
+
+configuration parse_args(int argc, char *argv[])
+{
+    configuration config;
+    po::positional_options_description po;
+    po.add("file", 1);
+
+    po::options_description desc("Allowed options");
+    desc.add_options()
+        ("help,h", po::bool_switch(&config.show_help), "show help")
+        ("tree,t", po::bool_switch(&config.show_tree), "show tree")
+        ("long,l", po::bool_switch(&config.show_long), "show long")
+        ("file", po::value<std::string>(&config.file_path), "file path")
+    ;
+
+    auto print_help = [&]
+    {
+        fs::path p(argv[0]);
+        std::cout << "Usage: " << p.filename().string() << " [OPTION]... [FILE]" << std::endl << std::endl;
+        std::cout << desc << std::endl;
+    };
+
+    try
+    {
+        po::variables_map vm;
+        po::store(po::command_line_parser(argc, argv).options(desc).positional(po).run(), vm);
+        po::notify(vm);
+
+        if (config.show_help)
+            print_help();
+    }
+    catch(boost::exception& ex)
+    {
+        print_help();
```

Comment:
This complexity is unnecessary. See my other comment: just let the exceptions flow.

Reply:
the intent of this catch is to print help message if the user passes wrong/unknown parameters. Changed `boost::exception` to `boost::program_options::error`. is that acceptable?